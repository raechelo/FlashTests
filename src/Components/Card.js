import React, { Component } from 'react';
import Q from './Q';
import M from './M';

export default class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
     currQ: {},
     ansBank: [],
     displayQ: false
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setState( {currQ: this.props.q, ansBank: this.props.shuffleAs} )
    }
  }

  render() {
    
    const multiclue = 
    <M 
    iq={this.props.iq} 
    qObj={this.state.currQ} 
    q={this.state.currQ.question} 
    a={this.state.currQ.answer} 
    ansBank={this.props.ansBank} 
    validA={this.checkAns}
    saveToStore={this.props.saveToStore}
    /> 

    const clue = 
    <Q 
    iq={this.props.iq} 
    qObj={this.state.currQ} 
    q={this.state.currQ.question} 
    a={this.state.currQ.answer} 
    validA={this.checkAns}
    saveToStore={this.props.saveToStore}
    />
    
      return (
        <div className="card">
          <div className="qCard">
          {this.state.currQ.type === 'Multi' ? multiclue : clue}
            <div className="card-btns">
              <button className="card-btn" onClick={this.props.nextQ}>Next Question</button>
              <button className="card-btn" onClick={this.props.startG} >Start Game</button>
              <button className="card-btn" onClick={this.props.oldCards} >Study Old Cards</button>
            </div>
          </div>
        </div>
      )
  }
}