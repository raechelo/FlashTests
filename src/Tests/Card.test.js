import React from 'react';
import Card from '../Components/Card';
import { shallow } from 'enzyme';

describe('Card', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card />
    );
    
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper default state', () => {
    expect(wrapper.state()).toEqual( {currQ: {}, ansBank: [], displayQ: false} )
  })
})