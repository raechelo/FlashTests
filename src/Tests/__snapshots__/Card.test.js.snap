// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all the data passed in 1`] = `
<div
  className="card"
>
  <div
    className="qCard"
  >
    <Q
      qObj={Object {}}
    />
    <div
      className="card-btns"
    >
      <button
        className="card-btn"
      >
        Next Question
      </button>
      <button
        className="card-btn"
      >
        Start Game
      </button>
      <button
        className="card-btn"
      >
        Study Old Cards
      </button>
    </div>
  </div>
</div>
`;
